### PR TITLE
[4700] Support: add ability to delete a user

### PR DESCRIPTION
**Pull request by d-a-v-e**
### Context

Support and operations should be able to delete / remove a user from within the support ui.

### Changes proposed in this pull request

* A link is shown on the user page to 'Delete user'
* Pressing the link takes the support user to a confirmation page
* After confirming, the user is deleted.
* The support user is redirected to the users index with a success flash message.

### Notes

This does not delete the user, only discards (to maintain timeline properties)

### CODE REVIEW

**Review by markhallen — APPROVED** [review]
LGTM

**Review by edwardhorsford — CHANGES_REQUESTED** [review]
There are some issues with this page:
Screenshot 2022-09-22 at 11 37 26

Issues:
* Page layout is off - esp the back link - can we look at existing delete pages?
* Need correct apostrophes rather than primes
* Cancel link is too small - check how other cancel links are done

I got this weird case when deleting a user:
Screenshot 2022-09-22 at 11 38 52

I *think* this is because I deleted an admin account. I think this is a requirement we missed - we should not let support users delete admin accounts (I reckon only devs should be able to create or delete admin accounts).

**Comment by claire-hughez** [comment]
@edwardhorsford do you have any examples of the back links or cancel text that Dave can steal please?

Also, I can't view this now as Ed deleted the account to view things on system admin.... hahahhaha

**Comment by d-a-v-e** [comment]
> 

i've got the backlink sorted - it's wrapped in a breadcrumb component throughout the rest of the app

**Comment by d-a-v-e** [comment]
@edwardhorsford think all wrinkles are addressed. Can no longer delete Admins as well:


https://user-images.githubusercontent.com/6339025/191772493-2ffb0a78-455c-4320-97c8-2ebab8bdcb13.mov

**Comment by claire-hughez** [comment]
I still can't log in now to check this PR haha. How can we recreate the account here?

File: app/views/system_admin/users/delete.html.erb [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+<div class="govuk-grid-row">
+  <div class="govuk-grid-column-two-thirds-from-desktop">
+
+    <span class="govuk-caption-l"> <%= @user.name %> </span>
+    <h1 class="govuk-heading-l">
+      Are you sure you want to delete <%= @user.name %>’s account?
+    </h1>
+
+    <%= govuk_button_to("Yes I'm sure - delete #{@user.name}’s account", user_path(@user), method: :delete, class: "govuk-button--warning") %>
```
Comment (edwardhorsford) [inline app/views/system_admin/users/delete.html.erb:18]:
```suggestion
    <%= govuk_button_to("Yes I’m sure – delete #{@user.name}’s account", user_path(@user), method: :delete, class: "govuk-button--warning") %>
```

File: app/views/system_admin/users/delete.html.erb [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  <%= render GovukComponent::BackLinkComponent.new(
+        text: "Back",
+        href: user_path(@user),
+      ) %>
+<% end %>
+
+<div class="govuk-grid-row">
+  <div class="govuk-grid-column-two-thirds-from-desktop">
+
+    <span class="govuk-caption-l"> <%= @user.name %> </span>
```
Comment (edwardhorsford) [inline app/views/system_admin/users/delete.html.erb:13]:
```suggestion
    <span class="govuk-caption-l"><%= @user.name %></span>
```

**Review by edwardhorsford — CHANGES_REQUESTED** [review]

**Review by edwardhorsford — APPROVED** [review]

**Comment by claire-hughez** [comment]
Looks perfect to me Dave!

**Pull request merged**